dev-2/dev-2.tests/StringAnalyzerTests.cs

```diff
@@ -0,0 +1,209 @@
+        {
+            //Arrange
+            string testString = "aaaa";
+            StringAnalyzer analyzer = new StringAnalyzer("");
+
+            //Act
+            bool result = analyzer.IsSequenceRepetitiveOnly(testString);
+
+            //Assert
+            Assert.IsTrue(result);
+        }
+
+        [TestMethod]
+        public void RepetitiveString_aaab_result_false()
+        {
+            //Arrange
+            string testString = "aaab";
+            StringAnalyzer analyzer = new StringAnalyzer("");
+
+            //Act
+            bool result = analyzer.IsSequenceRepetitiveOnly(testString);
+
+            //Assert
+            Assert.IsFalse(result);
+        }
+        #endregion
+
+        #region UniqueOnlyMethod
+        [TestMethod]
+        public void UniqueString_abcd_result_true()
+        {
+            //Arrange
+            string testString = "abcd";
+            StringAnalyzer analyzer = new StringAnalyzer("");
+
+            //Act
+            bool result = analyzer.IsSequenceUniqueOnly(testString);
+
+            //Assert
+            Assert.IsTrue(result);
+        }
+
+        [TestMethod]
+        public void UniqueString_abcc_result_false()
+        {
+            //Arrange
+            string testString = "abcc";
+            StringAnalyzer analyzer = new StringAnalyzer("");
+
+            //Act
+            bool result = analyzer.IsSequenceUniqueOnly(testString);
+
+            //Assert
+            Assert.IsFalse(result);
+        }
+        #endregion
+
+        #region LongestUniqueSubstringMethod
+        [TestMethod]
+        public void LongestUniqueSubstring_aaaa_result_a()
+        {
+            //Arrange
+            string testString = "aaaa";
+            string expected = "a";
+            StringAnalyzer analyzer = new StringAnalyzer(testString);
+
+            //Act
+            string result = analyzer.GetUniqueSymbolSequence();
+
+            //Assert
+            Assert.AreEqual(expected, result);
+        }
+
+        [TestMethod]
+        public void LongestUniqueSubstring_aaba_result_ab()
+        {
+            //Arrange
+            string testString = "aaba";
+            string expected = "ab";
+            StringAnalyzer analyzer = new StringAnalyzer(testString);
+
+            //Act
+            string result = analyzer.GetUniqueSymbolSequence();
+
+            //Assert
+            Assert.AreEqual(expected, result);
+        }
+
+        [TestMethod]
+        public void LongestUniqueSubstring_abcd_result_abcd()
+        {
+            //Arrange
+            string testString = "abcd";
+            string expected = "abcd";
+            StringAnalyzer analyzer = new StringAnalyzer(testString);
+
+            //Act
+            string result = analyzer.GetUniqueSymbolSequence();
+
+            //Assert
+            Assert.AreEqual(expected, result);
+        }
+#endregion
+
+        #region LongestRepetitiveDigitSubstringMethod
+        [TestMethod]
```

Comment:
Use more common names for test methods- negative, positive, Better LongestRepetitiveDigitSubstringFromLettersStringNegative- do not use details of test in the name - string like "aaaa" in a name not a good idea